qemu/tests/cpu_flag_intel_pt.py

```diff
@@ -0,0 +1,64 @@
+import logging
+
+from avocado.utils import process
+from virttest import error_context, env_process
+from provider.cpu_utils import check_cpu_flags
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Test cpu flag intel-pt.
+    1) Check if current flags are in the supported lists, if no, cancel test
+    2) Otherwise, set pt_model = 1 on host at first
+    3) Boot guest with cpu model 'host' without intel-pt.
+    4) Check cpu flags in guest(only for linux guest)
+    5) For q35
+       5.1) boot guest with cpu model with intel-pt
+    6) For pc
+       6.1) boot guest with intel-pt and min-level=0x14
+    7) Check cpu flags in guest(only for linux guest)
+    8) Restore pt_model value on host at last
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    flags = params["flags"]
+    check_host_flags = params.get_boolean("check_host_flags")
+    if check_host_flags:
+        check_cpu_flags(params, flags, test)
+
+    set_pt_model = params.get("set_pt_model")
+    check_pt_model = params.get("check_pt_model")
+    origin_value = process.getoutput(check_pt_model).strip()
+
+    try:
+        if origin_value != '1':
+            process.system(set_pt_model % '1', shell=True)
+        pt_model = process.getoutput(check_pt_model).strip()
+        if pt_model != '1':
+            test.cancel("pt_model can't be set to 1")
+
+        for x in range(2):
+            if x >= 1:
+                params["auto_cpu_model"] = 'yes'
+                if params["machine_type"] == 'q35':
+                    params["cpu_model_flags"] += ",intel_pt"
+                else:
+                    params["cpu_model_flags"] += ",intel_pt,min-level=0x14"
```

Comment:
Let's put the machine type difference in cfg.

Reply:
@huangyum Thanks and update, please help to review.